src/amo/components/Addon/index.js

```diff
@@ -331,6 +333,12 @@ export class AddonBase extends React.Component {
                   label={this.getFeaturedText(addonType)}
                 />
               ) : null}
+              {isRestartRequired ? (
```

Comment:
Yeah, I see what you're saying. I really want to throw that `addons` reducer out the window but let's do this. Let's continue with your approach. I'll add another review with some change requests to it.